histoqc/BaseImage.py

```diff
@@ -1,13 +1,28 @@
 import logging
 import os
 import numpy as np
-import inspect
 import zlib, dill
 from distutils.util import strtobool
+from PIL import Image
 
 #os.environ['PATH'] = 'C:\\research\\openslide\\bin' + ';' + os.environ['PATH'] #can either specify openslide bin path in PATH, or add it dynamically
 import openslide
+def getBestThumb(s, x, y, size):
+    osh = s["os_handle"]
+    if s["bounding_box"]:
```

Comment:
a small point here,  if the "else" code is quite small ( in this case 1 line) and includes a return statement, its better programming style to say:

```
if NOT condition
      return

-rest- of code
```

which eliminates the "else" statement entirely doesn't have a "dangling return" at the end of the function

it also reduces potential indentation errors and makes code readability/undrestanding a bit easier